kobo/apps/trash_bin/tests/storage_cleanup/test_attachment_cleanup.py

```diff
@@ -0,0 +1,217 @@
+                'effective_limit': 100000,
+                'balance_value': -82255,
+                'balance_percent': 182,
+                'exceeded': True
+            },
+        }
+        with patch(
+            'kobo.apps.subsequences.api_view.ServiceUsageCalculator.get_usage_balances',  # noqa
+            return_value=mock_balances,
+        ):
+            auto_delete_excess_attachments(self.owner.pk)
+            self.assertFalse(Attachment.objects.filter(pk=self.attachment.pk).exists())
+            self.assertEqual(Attachment.objects.all().count(), 0)
+
+        updated_balances = {
+            UsageType.ASR_SECONDS: None,
+            UsageType.MT_CHARACTERS: None,
+            UsageType.STORAGE_BYTES: {
+                'effective_limit': 100000,
+                'balance_value': 100000,
+                'balance_percent': 100,
+                'exceeded': False
+            },
+            UsageType.SUBMISSION: None
+        }
+
+        # Add another attachment after user is now within limit
+        instance_uid = uuid.uuid4()
+        submission = {
+            'q1': 'audio_conversion_test_image.jpg',
+            '_uuid': str(instance_uid),
+            '_attachments': [
+                {
+                    'download_url': f'http://testserver/{self.owner.username}/audio_conversion_test_image.jpg',  # noqa
+                    'filename': f'{self.owner.username}/audio_conversion_test_image.jpg',  # noqa
+                    'mimetype': 'image/jpeg',
+                },
+            ],
+            '_submitted_by': self.owner.username,
+        }
+        self.asset.deployment.mock_submissions([submission])
+        self.assertEqual(Attachment.objects.all().count(), 1)
+
+        with patch(
+            'kobo.apps.subsequences.api_view.ServiceUsageCalculator.get_usage_balances',  # noqa
+            return_value=updated_balances
+        ):
+            auto_delete_excess_attachments(self.owner.pk)
+
+        # The new attachment should *not* be deleted since user is no longer over limit
+        self.assertEqual(Attachment.objects.all().count(), 1)
+
+    @pytest.mark.skipif(
+        not settings.STRIPE_ENABLED, reason='Requires stripe functionality'
+    )
+    @override_config(AUTO_DELETE_ATTACHMENTS=True)
+    def test_schedule_cleanup_task_only_for_users_exceeding_grace_period(self):
+        """
+        Test that only users over their limit for more than the grace period are
+        scheduled for attachment cleanup
+        """
+        ExceededLimitCounter.objects.create(
+            user=self.owner,
+            limit_type=UsageType.STORAGE_BYTES,
+            days=config.LIMIT_ATTACHMENT_REMOVAL_GRACE_PERIOD + 1
+        )
+
+        with patch(
+            'kobo.apps.trash_bin.tasks.attachment.auto_delete_excess_attachments.delay'
+        ) as mock_task:
+            schedule_auto_attachment_cleanup_for_users()
+            mock_task.assert_called_once_with(self.owner.pk)
```

Comment:
We have only 1 user. Would be great to assert the `schedule_auto_attachment_cleanup_for_users ` does not run for the other user.